### ISSUE TITLE
Support Java 21 runtime

### ISSUE THREAD

**Issue by charlie-harvey** [aws/aws-codebuild-docker-images#676]
Could we add support for Java 21 runtime. Thanks.

**Comment by charlie-harvey** [comment]
Just like Lambda: https://github.com/aws/aws-lambda-base-images/issues/113

**Comment by nindza** [comment]
+1

**Comment by shorn** [comment]
My workaround for codebuild project created via CDK, for  a `buildImage: LinuxBuildImage.STANDARD_7_0` environment:

```
phases: {
  'pre_build': {
    commands: [
      'echo "Installing Amazon Corretto 21..."',
      'wget --no-verbose -O corretto-21.0.1.12.1.tar.gz https://corretto.aws/downloads/resources/21.0.1.12.1/amazon-corretto-21.0.1.12.1-linux-x64.tar.gz',
      'mkdir -p /tmp/corretto21',
      'tar -zxf corretto-21.0.1.12.1.tar.gz -C /tmp/corretto21',
      'export PATH=/tmp/corretto21/amazon-corretto-21.0.1.12.1-linux-x64/bin:$PATH',
      'export JAVA_HOME=/tmp/corretto21/amazon-corretto-21.0.1.12.1-linux-x64',
      'java -version',
    ]
  },
}
```

**Comment by rmanig** [comment]
Another option is to roll your own codebuild image:
```
FROM public.ecr.aws/amazonlinux/amazonlinux:2023

RUN dnf install -y java-21-amazon-corretto-devel \
    && dnf clean all

# install other stuff
```

**Comment by AdamBien** [comment]
With the support for Java 21 in Lambdas -- CodeBuild should also ship with Java 21 out-of-the-box.

**Comment by charlie-harvey** [comment]
2 pizza teams that only communicate via API calls. There isn't always a
grand plan to move lots of things at once at Amazon. So "should" is a
really strong word. 😉

**Comment by AdamBien** [comment]
> 2 pizza teams that only communicate via API calls. There isn't always a grand plan to move lots of things at once at Amazon. So "should" is a really strong word. 😉

1. The client perspective: CodeBuild *has* to support Java 21 and should be synchronized with Lambda and other services. End users / AWS clients must not care how AWS is internally organised. :-)

2. The developer perspective: it *would be nice* to have Java 21 support in near future. I fully understand the complexities. Also they might be internal processes in place which make releases harder than expected.

**Comment by Dylan-AWS** [comment]
We released Java 21 in the AL:5.0 image and Ubuntu 7.0 (https://github.com/aws/aws-codebuild-docker-images/pull/689).

**Comment by Dylan-AWS** [comment]
CodeBuild released Corretto 21 for ARM (#697)
